CGAL 6: Incompatible changes to AABB_tree

I've just tried to compile bonsai against CGAL 6 (gcc 14.2.1). I've had to switch to a new [sfcgal](https://gitlab.com/sfcgal/SFCGAL/-/commits/master/?ref_type=HEADS) version and thus noticed that there are [incompatible changes](https://github.com/CGAL/cgal/commit/006d46b1d4cc8188992a04d06fb23020cd70d0d6) in https://github.com/CGAL/cgal/blob/master/AABB_tree/include/CGAL/AABB_traits.h :
```
[  6%] Building CXX object CMakeFiles/geometry_kernel_cgal.dir./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp.o
In file included from ./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.h:46,
                 from ./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp:22:
./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp: In member function 'bool ifcopenshell::geometry::kernels::CgalKernel::preprocess_boolean_operand(const IfcUtil::IfcBaseClass*, const std::__cxx11::list<CGAL::Polyhedron_3<CGAL::Epeck> >&, const std::__cxx11::list<CGAL::Nef_polyhedron_3<CGAL::Epeck> >&, const std::__cxx11::list<CGAL::Plane_3<CGAL::Epeck> >&, const cgal_shape_t&, CGAL::Nef_polyhedron_3<CGAL::Epeck>&, boolean_operand_preprocess)':
./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp:1300:31: error: 'AABB_traits' in namespace 'CGAL' does not name a template type; did you mean 'AABB_traits_3'?
 1300 |                 typedef CGAL::AABB_traits<Kernel_, AABB_face_graph_primitive>               AABB_face_graph_traits;
      |                               ^~~~~~~~~~~
      |                               AABB_traits_3
./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp:1302:33: error: 'AABB_face_graph_traits' was not declared in this scope; did you mean 'AABB_face_graph_primitive'?
 1302 |                 CGAL::AABB_tree<AABB_face_graph_traits> tree;
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
      |                                 AABB_face_graph_primitive
./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp:1302:55: error: template argument 1 is invalid
 1302 |                 CGAL::AABB_tree<AABB_face_graph_traits> tree;
      |                                                       ^
./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp:1308:71: error: no matching function for call to 'build_AABB_tree(CGAL::Polyhedron_3<CGAL::Epeck>&, int&)'
 1308 |                         CGAL::Polygon_mesh_processing::build_AABB_tree(tm, tree);
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
In file included from ./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp:35:
/usr/include/CGAL/Polygon_mesh_processing/locate.h:1523:1: note: candidate: 'template<class TriangleMesh, class Point3VPM, class NamedParameters> void CGAL::Polygon_mesh_processing::build_AABB_tree(const TriangleMesh&, CGAL::AABB_tree<CGAL::AABB_traits_3<typename CGAL::GetGeomTraits<TriangleMesh, NamedParameters>::type, CGAL::AABB_face_graph_triangle_primitive<TriangleMesh, Point3VPM> > >&, const NamedParameters&)'
 1523 | build_AABB_tree(const TriangleMesh& tm,
      | ^~~~~~~~~~~~~~~
/usr/include/CGAL/Polygon_mesh_processing/locate.h:1523:1: note:   template argument deduction/substitution failed:
./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp:1308:71: note:   mismatched types 'CGAL::AABB_tree<CGAL::AABB_traits_3<typename CGAL::GetGeomTraits<TriangleMesh, NamedParameters>::type, CGAL::AABB_face_graph_triangle_primitive<TriangleMesh, Point3VPM> > >' and 'int'
 1308 |                         CGAL::Polygon_mesh_processing::build_AABB_tree(tm, tree);
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp:1338:105: error: no matching function for call to 'locate_with_AABB_tree(CGAL::HalfedgeDS_vertex_base<CGAL::HalfedgeDS_list_types<CGAL::Epeck, CGAL::I_Polyhedron_derived_items_3<CGAL::Polyhedron_items_3>, std::allocator<int> >, std::integral_constant<bool, true>, CGAL::Point_3<CGAL::Epeck> >::Point&, int&, CGAL::Polyhedron_3<CGAL::Epeck>&, CGAL::Named_function_parameters<double, CGAL::internal_np::snapping_tolerance_t, CGAL::internal_np::No_property>)'
 1338 |                                         auto ploc = CGAL::Polygon_mesh_processing::locate_with_AABB_tree(it->point(), tree, tm, CGAL::Polygon_mesh_processing::parameters::snapping_tolerance(1.e-5));
      |                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/CGAL/Polygon_mesh_processing/locate.h:1603:1: note: candidate: 'template<class TriangleMesh, class Point3VPM, class NamedParameters> typename CGAL::Polygon_mesh_processing::internal::Location_traits<TriangleMesh, NamedParameters>::Face_location CGAL::Polygon_mesh_processing::locate_with_AABB_tree(const typename internal::Location_traits<TriangleMesh, NamedParameters>::Point&, const CGAL::AABB_tree<CGAL::AABB_traits_3<typename CGAL::GetGeomTraits<TriangleMesh, NamedParameters>::type, CGAL::AABB_face_graph_triangle_primitive<TriangleMesh, Point3VPM> > >&, const TriangleMesh&, const NamedParameters&)'
 1603 | locate_with_AABB_tree(const typename internal::Location_traits<TriangleMesh, NamedParameters>::Point& p,
      | ^~~~~~~~~~~~~~~~~~~~~
/usr/include/CGAL/Polygon_mesh_processing/locate.h:1603:1: note:   template argument deduction/substitution failed:
./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp:1338:105: note:   mismatched types 'const CGAL::AABB_tree<CGAL::AABB_traits_3<typename CGAL::GetGeomTraits<TriangleMesh, NamedParameters>::type, CGAL::AABB_face_graph_triangle_primitive<TriangleMesh, Point3VPM> > >' and 'int'
 1338 |                                         auto ploc = CGAL::Polygon_mesh_processing::locate_with_AABB_tree(it->point(), tree, tm, CGAL::Polygon_mesh_processing::parameters::snapping_tolerance(1.e-5));
      |                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/CGAL/Polygon_mesh_processing/locate.h:1778:1: note: candidate: 'template<class TriangleMesh, class Point3VPM, class NamedParameters> typename CGAL::Polygon_mesh_processing::internal::Location_traits<TriangleMesh, NamedParameters>::Face_location CGAL::Polygon_mesh_processing::locate_with_AABB_tree(const typename internal::Location_traits<TriangleMesh, NamedParameters>::Ray&, const CGAL::AABB_tree<CGAL::AABB_traits_3<typename CGAL::GetGeomTraits<TriangleMesh, NamedParameters>::type, CGAL::AABB_face_graph_triangle_primitive<TriangleMesh, Point3VPM> > >&, const TriangleMesh&, const NamedParameters&)'
 1778 | locate_with_AABB_tree(const typename internal::Location_traits<TriangleMesh, NamedParameters>::Ray& ray,
      | ^~~~~~~~~~~~~~~~~~~~~
/usr/include/CGAL/Polygon_mesh_processing/locate.h:1778:1: note:   template argument deduction/substitution failed:
./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp:1338:105: note:   mismatched types 'const CGAL::AABB_tree<CGAL::AABB_traits_3<typename CGAL::GetGeomTraits<TriangleMesh, NamedParameters>::type, CGAL::AABB_face_graph_triangle_primitive<TriangleMesh, Point3VPM> > >' and 'int'
 1338 |                                         auto ploc = CGAL::Polygon_mesh_processing::locate_with_AABB_tree(it->point(), tree, tm, CGAL::Polygon_mesh_processing::parameters::snapping_tolerance(1.e-5));
      |                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/geometry_kernel_cgal.dir/build.make:90: CMakeFiles/geometry_kernel_cgal.dir./IfcOpenShell/src/ifcgeom/kernels/cgal/CgalKernel.cpp.o] Error 1

```

--- a/src/ifcgeom/kernels/cgal/CgalKernel.cpp
+++ b/src/ifcgeom/kernels/cgal/CgalKernel.cpp
@@ -1315,7 +1315,11 @@ bool CgalKernel::preprocess_boolean_operand(const IfcUtil::IfcBaseClass* log_ref
 	if (proc == PP_SNAP_POINTS_TO_FIRST_OPERAND) {
 		static int NN = 0;
 		typedef CGAL::AABB_face_graph_triangle_primitive<cgal_shape_t>                AABB_face_graph_primitive;
-		typedef CGAL::AABB_traits<Kernel_, AABB_face_graph_primitive>               AABB_face_graph_traits;
+#if CGAL_VERSION_NR >= 1060000000
+	    typedef CGAL::AABB_traits_3<Kernel_, AABB_face_graph_primitive>               AABB_face_graph_traits;
+#else
+	    typedef CGAL::AABB_traits<Kernel_, AABB_face_graph_primitive>               AABB_face_graph_traits;
+#endif
 
 		CGAL::AABB_tree<AABB_face_graph_traits> tree;
 

